hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -119,18 +125,32 @@ public void doStateTransitionOperation(final TransactionBody contractCallTxn, fi
 				: Bytes.EMPTY;
 
 		// --- Do the business logic ---
-		if (incrementCounter) {
+		if (relayerId != null) {
```

Comment:
nit: you could move the contents of this if to the `TransactionProcessingResult` assigning else below for simplicity